flexget/utils/titles/movie.py

```diff
@@ -76,10 +79,17 @@ def parse(self, data=None):
             data = self.data
 
         # Move anything in leading brackets to the end
-        data = re.sub(r'^\[(.*?)\](.*)', r'\2 \1', data)
+        prefix = re.match(r'^(\[(?:.*?)\])(?:.*)', data)
+        prefix = prefix.group(1) if prefix else ''
+        data = re.sub(r'^(\[(?:.*?)\])(.*)', r'\2', data)
 
-        for char in '[]()_,.':
+        for char in '()_,.':
             data = data.replace(char, ' ')
+            prefix = prefix.replace(char, ' ')
+        data = data.replace('[', ' ')
+        data = data.replace(']', ' ')
+
+        data += prefix
```

Comment:
Can we clean this up a bit, and update the comments to reflect what's happening? I think it's splitting a bracketed prefix, cleaning the 2 parts separately, then rejoining, no?